pkg/hostap/monitor.go

```diff
@@ -240,34 +264,61 @@ func (m *InterfaceMonitor) handleIfEvents() {
 			close(m.ifEventCh)
 			return
 		case linkUpdateEvent := <-m.ifEventCh:
-			newOpState := linkUpdateEvent.Link.Attrs().OperState
-			if newOpState == m.operState {
-				level.Info(m.Logger).Log("interface", "event", m.IfName, "op state not changed", m.operState)
-				continue
+			if linkUpdateEvent.Link.Attrs().Name == m.IfName {
+				m.handlePfEvent(linkUpdateEvent)
+			} else {
+				m.handleVfEvent(linkUpdateEvent)
```

Comment:
++

I love how clean the code gets when we pull more complicated if clauses out into separate functions!